File: System/Morality.cs
// File = robjam1990/Psychosis/Gameplay/System/Morality.cs

using System;
using System.Collections.Generic;

// Define Morality class
public class Morality
{
    private string jurisdiction = "local"; // Default jurisdiction for morality spectrum
    private Dictionary<string, double> actions = new Dictionary<string, double>(); // Dictionary to store actions and their morality consequences

    // Set the jurisdiction for morality spectrum
    public void SetJurisdiction(string jurisdiction)
    {
        this.jurisdiction = jurisdiction;
    }

    // Add an action with its morality consequences
    public void AddAction(string action, double moralityConsequence)
    {
        actions[action] = moralityConsequence;
    }

    // Get morality consequence for a specific action
    public double GetMoralityConsequence(string action)
    {
        return actions.ContainsKey(action) ? actions[action] : 0;
    }

    // Calculate morality based on actions performed
    public double CalculateMorality(List<string> actionsPerformed)
    {
        double morality = 0;

        foreach (string action in actionsPerformed)
        {
            if (actions.ContainsKey(action))
            {
                morality += actions[action];
            }
        }

        // Apply consequence multiplier if morality is below 50
        if (morality < 50)
        {
            morality *= 1.5;
        }

        return morality;
    }

    // Update morality jurisdiction
    public void UpdateJurisdiction(string newJurisdiction)
    {
        jurisdiction = newJurisdiction;
    }
}

// Example usage:
class Program
{
    static void Main(string[] args)
    {
        // Instantiate Morality object
        Morality moralitySystem = new Morality();

        // Set jurisdiction
        moralitySystem.SetJurisdiction("local");

        // Add actions and their morality consequences
        moralitySystem.AddAction("helping villagers", 0.01); // Positive morality consequence
        moralitySystem.AddAction("stealing from others", -0.02); // Negative morality consequence

        // Get morality consequence for a specific action
        Console.WriteLine(moralitySystem.GetMoralityConsequence("stealing from others")); // Output: -0.02

        // Calculate morality based on actions performed
        List<string> actionsPerformed = new List<string> { "helping villagers", "stealing from others" };
        Console.WriteLine(moralitySystem.CalculateMorality(actionsPerformed)); // Output: -0.015

        // Update morality jurisdiction
        moralitySystem.UpdateJurisdiction("global");
        Console.WriteLine(moralitySystem.GetJurisdiction()); // Output: global
    }
}

File: System/Morality.py
# File = robjam1990/Psychosis/Gameplay/System/Morality.py

# Define Morality class
class Morality:
    def __init__(self):
        self.jurisdiction = "local"  # Default jurisdiction for morality spectrum
        self.actions = {}  # Dictionary to store actions and their morality consequences

    # Set the jurisdiction for morality spectrum
    def set_jurisdiction(self, jurisdiction):
        self.jurisdiction = jurisdiction

    # Add an action with its morality consequences
    def add_action(self, action, morality_consequence):
        self.actions[action] = morality_consequence

    # Get morality consequence for a specific action
    def get_morality_consequence(self, action):
        return self.actions.get(action)

    # Calculate morality based on actions performed
    def calculate_morality(self, actions_performed):
        morality = 0

        for action in actions_performed:
            if action in self.actions:
                morality += self.actions[action]

        # Apply consequence multiplier if morality is below 50
        if morality < 50:
            morality *= 1.5

        return morality

    # Update morality jurisdiction
    def update_jurisdiction(self, new_jurisdiction):
        self.jurisdiction = new_jurisdiction


# Example usage:
# Instantiate Morality object
morality_system = Morality()

# Set jurisdiction
morality_system.set_jurisdiction("local")

# Add actions and their morality consequences
morality_system.add_action("helping villagers", 0.01)  # Positive morality consequence
morality_system.add_action("stealing from others", -0.02)  # Negative morality consequence

# Get morality consequence for a specific action
print(morality_system.get_morality_consequence("stealing from others"))  # Output: -0.02

# Calculate morality based on actions performed
actions_performed = ["helping villagers", "stealing from others"]
print(morality_system.calculate_morality(actions_performed))  # Output: -0.015

# Update morality jurisdiction
morality_system.update_jurisdiction("global")
print(morality_system.jurisdiction)  # Output: global
